test/test_torch.py

```diff
@@ -10779,9 +10779,32 @@ def test_tensor_factories_empty(self, device):
         self.assertEqual((0,), torch.bartlett_window(0, periodic=False, device=device).shape)
         self.assertEqual((0,), torch.hamming_window(0, device=device).shape)
         self.assertEqual((0,), torch.hann_window(0, device=device).shape)
+        self.assertEqual((0,), torch.vandermonde_matrix(torch.tensor([]), 0, device=device).shape)
         self.assertEqual((1, 1, 0), torch.tensor([[[]]], device=device).shape)
         self.assertEqual((1, 1, 0), torch.as_tensor([[[]]], device=device).shape)
 
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    @dtypes(torch.long, torch.double)
+    def test_vandermonde_matrix(self, device, dtype):
+        x = [1, 2, 3, 5]
+        n = 3
+
+        np_x = np.array(x, dtype=torch_to_numpy_dtype_dict[dtype])
+        pt_x = torch.tensor(x, dtype=dtype, device=device)
+
+        self.assertEqual(
+            torch.vandermonde_matrix(pt_x),
+            torch.from_numpy(np.vander(np_x)))
+
+        self.assertEqual(
```

Comment:
Good testing of variations. When adding multiple n you can probably loop over the values. Same with true/false (you can toggle it for each configuration).